packages/central-server/src/apiV2/tasks/CreateTask.js

```diff
@@ -21,13 +21,15 @@ export class CreateTask extends CreateHandler {
   }
 
   async createRecord() {
-    const { comment, ...data } = this.newRecordData;
-
-    return this.models.wrapInTransaction(async transactingModels => {
-      const task = await transactingModels.task.create(data);
+    const { comment, ...newRecordData } = this.newRecordData;
+    await this.models.wrapInTransaction(async transactingModels => {
+      const task = await transactingModels.task.create(newRecordData);
+      // Add the user comment first, since the transaction will mean that all comments have the same created_at time, but we want the user comment to be the 'most recent'
       if (comment) {
-        await task.addComment(comment, this.req.user.id);
+        await task.addComment(comment, this.req.user.id, 'user');
       }
+      await task.addComment('Created this task', this.req.user.id, 'system');
```

Comment:
The task_comment types seem a bit like magic strings. It's a shame that we don't really have typescript for these but it would be nice to create an enum for them somehow. One approach which I've seen in the Entity model is to just add a property on the model and then reference it from there which I think is pretty nice. eg. `models.entity.types.FACILITY`

Reply:
I see that we also import from the types package in the central server so that could be another option.